src/windshaft/map-serializer/anonymous-map-serializer/analysis-serializer.js

```diff
@@ -1,55 +1,35 @@
 var _ = require('underscore');
-var LayerTypes = require('../../../geo/map/layer-types.js');
+var AnalysisService = require('../../../analysis/analysis-service');
 
 /**
- * Return a payload with the serialization of all the analyses in the 
+ * Return a payload with the serialization of all the analyses in the
  * layersCollection and the dataviewsCollection.
  */
 function serialize (layersCollection, dataviewsCollection) {
-  var layerAnalyses = _getAnalysesFromLayers(layersCollection);
-  var dataviewsAnalyses = _getAnalysesFromDataviews(dataviewsCollection);
-  return _generateUniqueAnalysisList(layerAnalyses.concat(dataviewsAnalyses));
+  var analysisList = AnalysisService.getAnalysisList(layersCollection, dataviewsCollection);
+  return _generateUniqueAnalysisList(analysisList);
 }
 
 /**
  * Return an analysis list without duplicated or nested analyses
  */
-function _generateUniqueAnalysisList (analysesList) {
-  var analysesIds = {};
-  return _.reduce(analysesList, function (list, analysis) {
-    if (!analysesIds[analysis.get('id')] && !_isAnalysisPartOfOtherAnalyses(analysis, analysesList)) {
-      analysesIds[analysis.get('id')] = true; // keep a set of already added analysis.
+function _generateUniqueAnalysisList (analysisList) {
+  var analysisIds = {};
+  return _.reduce(analysisList, function (list, analysis) {
+    if (!analysisIds[analysis.get('id')] && !_isAnalysisPartOfOtherAnalyses(analysis, analysisList)) {
```

Comment:
we can use `analysis.isSourceOfAnyModel` instead `_isAnalysisPartOfOtherAnalyses` 

In fact, `_isAnalysisPartOfOtherAnalyses` could be removed

Reply:
I don't think that would work @IagoLast. `analysis.isSourceOfAnyModel` checks if the analysis is referenced by a `layer` or `dataview`. `_isAnalysisPartOfOtherAnalyses` is checking if the analysis is part of another analysis.

Reply:
I just change the analysisList name in this function, so may be we can discuss changing this method in other PR